src/shared/components/oncoprint/controls/OncoprintControls.tsx

```diff
@@ -694,8 +672,32 @@ export default class OncoprintControls extends React.Component<
 
     @autobind
     @action
-    private onSelectGenericAssayEntities(selectedElements: ISelectOption[]) {
-        this._selectedGenericAssayEntityIds = selectedElements.map(o => o.id);
+    private onSelectGenericAssayEntities(
+        selectedOptions: ISelectOption[],
+        selectInfo: any
+    ) {
+        // selectedOptions can be null if delete the last selected option
+        let candidateOptions = selectedOptions ? selectedOptions : [];
+        // if choose select all option, add all filtered options
+        if (
+            selectInfo.action === 'select-option' &&
+            selectInfo.option.id === 'select_all_filtered_options'
+        ) {
+            candidateOptions = _.union(
```

Comment:
why do you do this union?  is it because there maybe previously selected options that you want to retain?  if so, that should be explained in comment